src/Microsoft.TemplateEngine.Edge/BuiltInManagedProvider/GlobalSettingsTemplatePackageProvider.cs

```diff
@@ -102,6 +102,9 @@ public async Task<IReadOnlyList<CheckUpdateResult>> GetLatestVersionsAsync(IEnum
             {
                 result.AddRange(task.Result);
             }
+
+            await UpdateTemplatePackageMetadataAsync(result.Select(r => r.TemplatePackage), cancellationToken).ConfigureAwait(false);
```

Comment:
currently this API is used with all packages by SDK, but it theory it may be used only for specific set.
Therefore the update should happen only for matching packages, and not overwrite the whole content of package.json.